Zend/zend_types.h

```diff
@@ -141,8 +142,11 @@ typedef struct {
 	zend_type types[1];
 } zend_type_list;
 
-#define _ZEND_TYPE_EXTRA_FLAGS_SHIFT 25
-#define _ZEND_TYPE_MASK ((1u << 25) - 1)
+#define _ZEND_TYPE_EXTRA_FLAGS_SHIFT 26
+#define _ZEND_TYPE_MASK ((1u << 26) - 1)
+// Used to signify that type.ptr is not a `zend_string*` but a `const char*`,
+// in which case _ZEND_TYPE_NAME_BIT must also be set.
+#define _ZEND_TYPE_LITERAL_NAME_BIT (1u << 25)
 /* Only one of these bits may be set. */
 #define _ZEND_TYPE_NAME_BIT (1u << 24)
 #define _ZEND_TYPE_LIST_BIT (1u << 22)
```

Comment:
You could move the NAME_BIT to use ``1u << 23`` and then have the LITERAL_NAME bit be ``1u << 24``

However, I wonder if it doesn't make more sense to have ``_ZEND_TYPE_NAME_BIT`` and ``_ZEND_TYPE_LITERAL_NAME_BIT`` be exclusive to each other? So that ``ZEND_TYPE_HAS_NAME()`` ensures that one has a ``zend_string*`` name.

Reply:
Or just use ``1u << 23`` for the LITERAL_NAME one actually

Reply:
Ah good catch, I completely missed the hole in the bitfield !

Reply:
> However, I wonder if it doesn't make more sense to have `_ZEND_TYPE_NAME_BIT` and `_ZEND_TYPE_LITERAL_NAME_BIT` be exclusive to each other?

Indeed, makes total sense, especially since `char*`s are supposed to be temporary and disappear completely once the arginfos are registered.

Reply:
So, I just tried that and it turns out to be less backwards compatible, causing some tests to fail... Need to investigate further...